### PR TITLE
Branks42 Email Opt In

### PR DESCRIPTION
**Pull request by branks42**
***Work Completed***
- New step in all modals to offer users the option to be notified of process completion via email
- Update saga, actions, api calls
- New components for all action modals

***Steps Required***
- N/A

### CODE REVIEW

File: src/components/download_stepper/DownloadStepperSubmit.js [hunk 1]
```diff
@@ -21,48 +18,39 @@ const CustomDownloadButton = withStyles({
...
   const targetToken = useSelector(state => state.apiTokens[state.selectedTarget.name]);
   const selectedResource = useSelector(state => state.selectedResource);
 
   const submitDownload = () => {
-    dispatch(actionCreators.download.downloadResource(selectedResource, targetToken, false));
+    dispatch(actionCreators.download.downloadResource(selectedResource, targetToken, false, emailValue));
     setActiveStep(prevActiveStep => prevActiveStep + 1);
   };
 
+  const searchKeystroke = (event) => {
```
Comment (bfox6) [inline src/components/download_stepper/DownloadStepperSubmit.js:32]: Change the name?

File: src/components/transfer/stepper/TransferStepperTransferButton.js [hunk 2]
```diff
@@ -52,107 +49,35 @@ export default function TransferStepperTransferButton({handleNext, selectedDupli
...
     handleNext()
   };
 
-  let destinationTargetReadableName = '';
-  for (let i = 0; i < available.length; i++) {
-    if (available[i].name === transferDestinationTarget) {
-      destinationTargetReadableName = available[i].readable_name
-    }
-  }
+  const searchKeystroke = (event) => {
```
Comment (bfox6) [inline src/components/transfer/stepper/TransferStepperTransferButton.js:59]: Change name

File: src/components/upload_stepper/UploadAgreement.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      backgroundColor: '#0a4996',
+    },
+  },
+})(Button);
+
+/**
+ * Component responsible for rendering the upload button in the upload stepper and passing the
+ * selected file to the Upload API endpoint
+ **/
+export default function UploadAgreement({handleNext, resourceToUploadTo}) {
```
Comment (bfox6) [inline src/components/upload_stepper/UploadAgreement.js:26]: Need resourceToUploadTo param?

File: src/components/upload_stepper/UploadButton.js [hunk 4]
```diff
@@ -25,74 +22,69 @@ const CustomUploadButton = withStyles({
...
-                ? `Upload to the ${selectedTarget.readable_name} resource '${sourceResource.title}'.`
-                : `Upload to ${selectedTarget.readable_name} as a new project.`
-            }
-          />
-
-          {/* Metadata Statement*/}
-          <IconListItem
-            icon={<EditIcon />}
-            text="Write or edit File Transfer Service Metadata file at the top level." />
+  const searchKeystroke = (event) => {
```
Comment (bfox6) [inline src/components/upload_stepper/UploadButton.js:58]: Change name

**Review by bfox6 — CHANGES_REQUESTED** [review]

File: src/components/upload_stepper/UploadAgreement.js (hunk 3 above)
Comment (branks42) [reply src/components/upload_stepper/UploadAgreement.js:26]: Yep, to determine the text to display in the actions list

**Pull request merged**